extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzBuildTimeConfig.java

```diff
@@ -57,6 +58,13 @@ public class QuartzBuildTimeConfig {
     @ConfigItem(defaultValue = "QRTZ_")
     public String tablePrefix;
 
+    /**
+     * The maximum amount of time Quarkus will wait for currently running jobs to finish.
+     * If the value is {@code 0}, then Quarkus will not wait at all for these jobs.
+     */
+    @ConfigItem(defaultValue = "PT10S")
+    public Duration shutdownWaitTime;
```

Comment:
This could be a runtime configuration.

Reply:
I think we should keep it build time until we someone describing a valid use case of why it should be runtime.

Reply:
I would add a reference to `org.quartz.Scheduler.shutdown(false)` so that it's clear what the `0` value does.

Reply:
Also we should probably mention that the default time unit is seconds...

Reply:
> Also we should probably mention that the default time unit is seconds...

We already have [this](https://quarkus.io/guides/all-config#duration-note-anchor-quarkus-all-config) in the all-configuration page that is used for all `Duration` types.

Reply:
I know but it's IMO less visible. Feel free to ignore my rants though ;-)

Reply:
I'd rather we don't add it, because then we would probably have to add it everywhere :)

Reply:
> I think we should keep it build time until we someone describing a valid use case of why it should be runtime.

I think it can be made as a runtime config  straight away and if someone needs to change the value then they get the benefit of runtime configs straight away as opposed to building their apps again.

Reply:
Let's let @mkouba break the tie :)

Reply:
> I'd rather we don't add it, because then we would probably have to add it everywhere :)

Yes +1, makes sense.

> Let's let @mkouba break the tie :)

+1 for a runtime config property. All [thread pool related config properties](https://quarkus.io/guides/quartz#quarkus-quartz_configuration) are runtime anyway.

Reply:
Okay, done